packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -1,20 +1,21 @@
 import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'tsutils';
 import * as ts from 'typescript';
 
 import * as util from '../util';
 
 type Options = [
   {
-    checkCompoundAssignments?: boolean;
     allowAny?: boolean;
+    allowBoolean?: boolean;
+    allowNullish?: boolean;
+    allowNumberAndString?: boolean;
```

Comment:
To me at least - this option seems like it circumvents the original intent of the rule - which is to enforce the two sides of the operator are the same type because TS doesn't enforce this.

TBH I don't know what the rule would check any more if you turned this option on - I think all of the remaining cases really would be TS errors.